src/AWS.Deploy.Common/TypeHintData/TypeHintResourceTable.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+
+namespace AWS.Deploy.Common.TypeHintData
+{
+    /// <summary>
+    /// Represents a list or table of existing AWS resources to allow selecting from
+    /// a list of existing resources when setting an OptionSettingItem
+    /// </summary>
+    public class TypeHintResourceTable
+    {
+        /// <summary>
+        /// Columns that should appear above the list of resources when presenting the
+        /// user a table or grid to select from
+        /// </summary>
+        /// <remarks>If this is null or empty, it implies that there is only a single column</remarks>
+        public List<TypeHintResourceColumn>? Columns { get; set; }
```

Comment:
I think we only need this to be a `List<string>?`

Reply:
I wanted to model the columns as classes as opposed to strings. That way if we add additional properties in the future (width, sort order, etc.), it'd be an additive change to an existing class as opposed to a breaking change for consumers of server mode to transition from `List<string>` to `List<SomeNewClass>`.

We could use a simpler model for the CLI-specific classes instead of the `/ServerMode/` ones, but I opted to keep them as similar as possible.